tox/__init__.py

```diff
@@ -1,89 +1,29 @@
-import os
-import signal
-
+import pluggy
 from pkg_resources import DistributionNotFound
 from pkg_resources import get_distribution
 
-from .hookspecs import hookimpl
+from .exc import exception  # noqa
 from .hookspecs import hookspec
 
+# NOTE: hookimpl and hookspec objects will be removed from API in tox 4 - see warning below
+__all__ = ('hookspec', 'hookimpl', 'cmdline', 'exception', '__version__')
+"""Everything explicitly exported here is part of the tox programmatic API.
+
+To override/modify tox behaviour also see tox.hookspec and its use with pluggy.
+
+"""
+
+# DEPRECATED - will go away in tox 4
```

Comment:
can we add some proxy thing for this to emit deprecation warning if imported instead of the deprecated comments? https://docs.python.org/2/library/exceptions.html#exceptions.DeprecationWarning

that way people using it will get some runtime noise to notify them 😄

Reply:
I had the same idea already, but was not immediately sure how to accomplish that given a module level object. I will definitely look into that. I want to wait first though for input from @RonnyPfannschmidt as he made a comment that needs further clarification (https://github.com/tox-dev/detox/pull/12#issuecomment-381348761). It might be that I got this part completely wrong and it needs a rework and undoing of a lot of PRs I opened - I really hope not, but life is unfair sometimes :D

Reply:
so found out https://www.python.org/dev/peps/pep-0562/ is for this, also shows a workaround for python 2 compatibility: ``Typical workarounds are assigning __class__ of a module object to a custom subclass of types.ModuleType or replacing the sys.modules item with a custom wrapper instance. ``